packages/auth/src/authClients/supabase.ts

```diff
@@ -1,30 +1,37 @@
-import type {
-  SupabaseAuthUser,
-  SupabaseClient as Supabase,
-  SupabaseAuthResponse,
-} from '@supabase/supabase-js'
+import { Session, User, Provider } from '@supabase/gotrue-js/dist/main/lib/types'
+import type SupabaseClient from '@supabase/supabase-js/dist/main/SupabaseClient'
 
 import type { AuthClient } from './index'
-export type SupabaseUser = SupabaseAuthUser
-export type { Supabase }
+export type Supabase = SupabaseClient
+export type SupabaseUser = User
 
 export interface AuthClientSupabase extends AuthClient {
   login(options: {
     email: string
     password: string
-  }): Promise<SupabaseAuthResponse>
+  }): Promise<{
+    data: Session | null;
```

Comment:
I am not a Typescript expert, but could one define a `SupabaseAuthResponse` and use it both here and in `signup` since `url` and `provider` are optional? Both share data, user, and error.

Reply:
Technically I could. But I don't want to. I think it would confuse people. Like "I wonder when this function would ever return `url` and `provider`?". What I'm doing here is exactly what supabase themselves are doing as well.